site/source/docs/porting/Audio.rst

```diff
@@ -0,0 +1,143 @@
+
+The above snippet usually works because most OpenAL implementations own and manage one or more separate threads. This is not the case in Emscripten.
+
+What you must do instead is perform each such query only once per "main loop iteration" (i.e the callback you provide via :c:func:`emscripten_set_main_loop` or :c:func:`emscripten_set_main_loop_arg`).
+
+
+Second: Avoid creating and destroying resources relentlessly.
+
+On some browsers (except, apparently, Firefox), garbage collection does not occur as much as it should. Knowing this, your app should ideally at most use one ``ALCcontext`` for playback and one ``ALCdevice`` for capture.
+
+Even if you're planning to suspend audio capture or playback for extended periods of time, it is probably better to keep contexts in memory (unlike in native apps where you should rather destroy them) to avoid risking memory leaks caused by poorly garbage-collected Javascript implementations.
+
+Of course, the current OpenAL implementation strives to be memory-efficient, but it has no control over lower-level mechanisms involved.
+
+
+.. _Audio-openal-capture-behavior-g:
+
+Emscripten-specific capture behavior
+====================================
+
+Attempting to open an input stream to the user's audio capture device
+results in the asynchronous appearance of a small browser-specific dialog
+asking for the user's permission.
+
+.. note:: 
+    On Firefox, the user is also given the opportunity to pick from a list of devices at that moment.
+    On other browsers, this is normally a simple "Allow/Deny" button pair on a toast.
+
+With this in mind, when ``alcCaptureOpenDevice()`` is called with valid and 
+supported parameters, a "proxy" device is returned, which successfully 
+captures no sample until the user clicks "Allow" on said dialog.
+
+That means, when calling ``alcGetIntegerv(device, ALC_CAPTURE_SAMPLES, 1, &nframes)``, ``nframes`` remains set to zero until the user clicks "Allow". You might want to make your app handle this specific behavior.
+
+If the user clicks "Deny", the device is invalidated (because this is somewhat
+similar to unplugging the physical device) and calls to ``alcCapture*`` functions on that
+device then consistently fail with ``ALC_INVALID_DEVICE``.
+Your application must be prepared to handle this properly.
+
+.. note::
+    Some browsers "remember" this choice and apply it automatically every time it would be asked again instead.
+    There's no way for the implementation to detect this behavior.
+
+
+.. _Audio-openal-capture-details-g:
+
+Useful implementation details of OpenAL capture
+===============================================
+
+Internally, Web Audio's capture data is always backed by a Javascript ``Float32Array``.  
+Thus, ``AL_FORMAT_MONO_FLOAT32`` and ``AL_FORMAT_STEREO_FLOAT32`` are the only formats which do not require running "type conversion passes" on acquired buffers.
```

Comment:
I had put them because it felt like this was a term I just pulled of nowhere and needed to show that I was aware of it. It's hard to put into words, but not important anymore since I'll change the wording to be more to the point, like : 
```
Thus, ``AL_FORMAT_MONO_FLOAT32`` and ``AL_FORMAT_STEREO_FLOAT32`` 
are the only formats which do not require converting acquired samples from their
initial type to another.
```